src/Mod/Path/PathScripts/PathOpGui.py

```diff
@@ -184,182 +190,225 @@ def setupContextMenu(self, vobj, menu):
         (see getSignalsForUpdate(obj)) this function can be used to register signal handlers
         manually.
-        Can safely be overwritten by subclasses.'''
+        Can safely be overwritten by subclasses."""
         # pylint: disable=unused-argument
         pass  # pylint: disable=unnecessary-pass
 
     def updateData(self, obj, prop):
-        '''updateData(obj, prop) ... overwrite if the receiver needs to react to property changes that might not have been caused by the receiver itself.
+        """updateData(obj, prop) ... overwrite if the receiver needs to react to property changes that might not have been caused by the receiver itself.
         Sometimes a model will recalculate properties based on a change of another property. In order to keep the UI up to date with such changes this
         function can be used.
         Please note that the callback is synchronous with the property assignment operation. Also note that the notification is invoked regardless of the
         actual value of the property assignment. In other words it also fires if a property gets assigned the same value it already has.
         Taking above observations into account the implementation has to take care that it doesn't overwrite modified UI values by invoking setFields(obj).
         This can happen if a subclass unconditionally transfers all values in getFields(obj) to the model and just calls setFields(obj) in this callback.
         In such a scenario the first property assignment will cause all changes in the UI of the other fields to be overwritten by setFields(obj).
-        You have been warned.'''
+        You have been warned."""
         # pylint: disable=unused-argument
         pass  # pylint: disable=unnecessary-pass
 
     def updateSelection(self, obj, sel):
-        '''updateSelection(obj, sel) ...
+        """updateSelection(obj, sel) ...
         overwrite to customize UI depending on current selection.
-        Can safely be overwritten by subclasses.'''
+        Can safely be overwritten by subclasses."""
         # pylint: disable=unused-argument
         pass  # pylint: disable=unnecessary-pass
 
     # helpers
+    # def selectInComboBox(self, name, combo):
+    #     """selectInComboBox(name, combo) ...
+    #     helper function to select a specific value in a combo box."""
+    #     index = combo.findText(name, QtCore.Qt.MatchFixedString)
+    #     if index >= 0:
+    #         combo.blockSignals(True)
+    #         combo.setCurrentIndex(index)
+    #         combo.blockSignals(False)
+
     def selectInComboBox(self, name, combo):
-        '''selectInComboBox(name, combo) ...
-        helper function to select a specific value in a combo box.'''
-        index = combo.findText(name, QtCore.Qt.MatchFixedString)
-        if index >= 0:
+        """selectInComboBox(name, combo) ...
+        helper function to select a specific value in a combo box."""
+        combo.blockSignals(True)
```

Comment:
Consider using `QSignalBlocker` instead, so you don't have to have all these block/unblock calls.